ee/vellum_ee/workflows/display/utils/tests/test_events.py

```diff
@@ -67,3 +68,106 @@ class TestWorkflow(BaseWorkflow):
 
     # AND workflow_version_exec_config is set to the expected config
     assert event.body.workflow_version_exec_config == expected_config
+
+
+class MockWorkflowDefinition(BaseWorkflow):
+    """Mock workflow definition for testing."""
+
+    pass
+
+
+@pytest.mark.parametrize(
+    "parent_context,expected_is_dynamic",
+    [
+        (None, False),
+        (
+            WorkflowDeploymentParentContext(
+                span_id=uuid4(),
+                deployment_id=uuid4(),
+                deployment_name="test-deployment",
+                deployment_history_item_id=uuid4(),
+                release_tag_id=uuid4(),
+                release_tag_name="test-tag",
+                workflow_version_id=uuid4(),
+                external_id=None,
+                metadata=None,
+                parent=None,
+            ),
+            False,
+        ),
+        (
+            WorkflowDeploymentParentContext(
+                span_id=uuid4(),
+                deployment_id=uuid4(),
+                deployment_name="test-deployment",
+                deployment_history_item_id=uuid4(),
+                release_tag_id=uuid4(),
+                release_tag_name="test-tag",
+                workflow_version_id=uuid4(),
+                external_id=None,
+                metadata=None,
+                parent=NodeParentContext(
+                    span_id=uuid4(),
+                    node_definition=MockWorkflowDefinition,
+                    parent=None,
+                ),
+            ),
+            True,
+        ),
+        (
+            NodeParentContext(
+                span_id=uuid4(),
+                node_definition=MockWorkflowDefinition,
+                parent=None,
+            ),
+            False,
+        ),
+        (
+            WorkflowDeploymentParentContext(
+                span_id=uuid4(),
+                deployment_id=uuid4(),
+                deployment_name="test-deployment",
+                deployment_history_item_id=uuid4(),
+                release_tag_id=uuid4(),
+                release_tag_name="test-tag",
+                workflow_version_id=uuid4(),
+                external_id=None,
+                metadata=None,
+                parent=WorkflowParentContext(
+                    span_id=uuid4(),
+                    workflow_definition=MockWorkflowDefinition,
+                    parent=None,
+                ),
+            ),
+            False,
+        ),
+    ],
+)
+def test_event_enricher_dynamic_marking_based_on_context(parent_context, expected_is_dynamic):
```

Comment:
delete this test, not emblematic of what we want to add

Reply:
in fact, just add test that asserts that a workflow executed as a subworkflow deployment asserts is_dynamic is true for now